python/ray/dag/tests/experimental/test_verify_graph.py

```diff
@@ -0,0 +1,232 @@
+# coding: utf-8
+import os
+import sys
+
+import pytest
+
+import ray
+import ray.cluster_utils
+from ray.experimental.channel.torch_tensor_type import TorchTensorType
+from ray.experimental.channel.conftest import start_nccl_mock
+from ray.tests.conftest import *  # noqa
+from ray.dag import InputNode, MultiOutputNode
+
+INVALID_GRAPH = "Detect a deadlock in the graph. If this is a false positive, "
+"please disable the graph verification by setting the environment "
+"variable RAY_ADAG_ENABLE_VERIFY_GRAPH to 0."
+
+
+@ray.remote(num_cpus=0, num_gpus=1)
+class MockedWorker:
+    def __init__(self):
+        pass
+
+    def start_mock(self):
+        """
+        Patch methods that require CUDA.
+        """
+        start_nccl_mock()
+
+    def no_op(self, value):
+        return value
+
+    def no_op_two(self, value1, value2):
+        return value1, value2
+
+
+@pytest.mark.parametrize("ray_start_regular", [{"num_gpus": 1}], indirect=True)
+def test_invalid_graph_1_actor(ray_start_regular):
```

Comment:
Might be nice to add a version of the DAG that does not have the NCCL edge (just uses the default transport), to check that the invalid graph assertion is not thrown for those DAGs.

Reply:
Updated https://github.com/ray-project/ray/pull/45960/commits/8588e0f68c4fc413b8b0df924bb4030c8023c729